Packet++/src/SmtpLayer.cpp

```diff
@@ -0,0 +1,248 @@
+#define LOG_MODULE PacketLogModuleSmtpLayer
+
+#include "SmtpLayer.h"
+
+#include <algorithm>
+
+namespace pcpp
+{
+	// ----------------- Class SmtpRequestLayer -----------------
+	bool SmtpRequestLayer::setCommand(SmtpCommand code) { return setCommandInternal(getCommandAsString(code)); }
+
+	SmtpRequestLayer::SmtpCommand SmtpRequestLayer::getCommand() const
+	{
+		size_t val = 0;
+		std::string field = getCommandString();
+
+		for (size_t idx = 0; idx < field.size(); ++idx)
+			val |= (field.c_str()[idx] << (idx * 8));
+
+		return static_cast<SmtpCommand>(val);
+	}
+
+	std::string SmtpRequestLayer::getCommandString() const { return getCommandInternal(); }
```

Comment:
Why can't we call `getCommandAsString(getCommand())` here? 🤔 

It'll be more consistent because in username/password packets the return value will be "Unknown command" and not the username/password value

Reply:
getCommand() also depends getCommandString(). So it will be like an infinite loop 😁

Reply:
Actually this function for the get command as a string not for returning explanation. I mean it returns "EHLO" etc.

Reply:
I understand your command now, updated!